src/github.com/matrix-org/dendrite/syncserver/storage/current_room_state_table.go

```diff
@@ -29,6 +29,8 @@ CREATE TABLE IF NOT EXISTS current_room_state (
 );
 -- for event deletion
 CREATE UNIQUE INDEX IF NOT EXISTS event_id_idx ON current_room_state(event_id);
+-- for querying membership states of users
+CREATE INDEX IF NOT EXISTS membership_idx ON current_room_state(membership) WHERE membership IS NOT NULL AND membership != 'leave';
```

Comment:
It might be desirable to include the `state_key` somewhere in the index, maybe the `type` as well?